src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -279,6 +290,22 @@ public RequestBuilder(
     this.host = hostName;
     this.userAgentSuffix = userAgentSuffix;
 
+    // create our security/token manager
+    if (credential instanceof KeyPair) {
+      securityManager =
+          new JWTManager(accountName, userName, (KeyPair) credential, telemetryService);
+      authType = Constants.JWT;
+    } else {
```

Comment:
Maybe do a switch statement here and throw an exception if not an instance of Keypair or OAuth.